src/bin/add/main.rs

```diff
@@ -143,6 +147,34 @@ fn handle_add(args: &Args) -> Result<()> {
     }
     let deps = &args.parse_dependencies()?;
 
+    if let Some(ref requested_features) = args.features {
+        let available_features = deps
+            .iter()
+            .map(|d| d.available_features.clone())
+            .flatten()
+            .collect::<Vec<String>>();
+        let flattened_features: Vec<String> = requested_features
+            .iter()
+            .map(|feat| {
+                feat.split(' ')
+                    .map(|s| s.trim().to_string())
+                    .collect::<Vec<String>>()
+            })
+            .flatten()
+            .collect();
+        let unknown_features: Vec<String> = flattened_features
```

Comment:
Can't these vec's be `&str` instead of `String` and avoid allocating memory until we need to report to the iser?

Reply:
Nit: If we use a `BTreeSet` for each of these, instead of a `Vec`, we can have it find the differences for us and keep them sorted.

Reply:
think I managed to address this in the new commit, but let me know if you're happy with the current solution